src/features/profile/profileSlice.ts

```diff
@@ -46,16 +46,18 @@ export const useLoggedInProfileUser = (username?: string) => {
     ({ profile }) => profile?.loggedInProfile?.user?.username
   );
 
-  if (query.isSuccess) {
-    // App should crash if profile DNE
-    if (!(query.data && query.data[0][0])) {
-      throw query.error;
+  useEffect(() => {
+    if (query.isSuccess) {
+      // App should crash if profile DNE
+      if (!(query.data && query.data[0][0])) {
+        throw query.error;
+      }
+      // Set the logged in profile once it loads
+      if (query.isSuccess && query.data[0][0].user.username !== profileUser) {
+        dispatch(setLoggedInProfile(query.data[0][0]));
+      }
     }
-    // Set the logged in profile once it loads
-    if (query.isSuccess && query.data[0][0].user.username !== profileUser) {
-      dispatch(setLoggedInProfile(query.data[0][0]));
-    }
-  }
+  }, [dispatch, profileUser, query.data, query.error, query.isSuccess]);
```

Comment:
fixing the "dispatching outside useEffect" issue again here

Reply:
See [this link](https://github.com/facebook/react/issues/18178) (again).